src/App.js

```diff
@@ -1,23 +1,80 @@
-import logo from './logo.svg';
 import './App.css';
+import Form from './components/Form';
+import ToDoList from './components/ToDoList';
+import { useState, useEffect } from 'react';
 
 function App() {
+  //State
+  const [inputText, setInputText] = useState('');
+  const [todos, setTodos] = useState([]);
+  const [status, setStatus] = useState('all');
+  const [filteredTodos, setFilteredTodos] = useState([]);
+  const [action, setAction] = useState('add');
+  const [itemToEdit, setItemToEdit] = useState(undefined);
```

Comment:
Lifting up state values is a good approach only for shared properties, in your case you persist almost all of your state values in parent component (thats same as keeping all of your values in global scope). Pls try reconsidering your state structure: `inputText` value should be in Form component, `action` should be in ToDoList component and so on. You should always keep loose coupling/high cohesion in your code

Reply:
fixed